src/compile/data/data.ts

```diff
@@ -149,46 +243,78 @@ export function parseLayerData(model: LayerModel): DataComponent {
 export function assembleData(model: Model, data: VgData[]) {
   const component = model.component.data;
 
-  const sourceData = source.assemble(model, component);
-  if (sourceData) {
-    data.push(sourceData);
+  if (!component) {
+    return data;
   }
 
-  summary.assemble(component, model).forEach(function(summaryData) {
-    data.push(summaryData);
-  });
+  let dataSource = source.assemble(model, component);
+  dataSource.name = model.dataName(SOURCE);
+  data.push(dataSource);
+
+  dataSource.transform = [].concat(
+    calculate.assemble(component),
+    filter.assemble(component),
+    nullFilter.assemble(component),
+    bin.assemble(component),
+    timeUnit.assemble(component)
+  );
+
+  if (component.includeRawDomain) {
```

Comment:
Probably need to create intermediate source if contains scale `sort`.

https://github.com/vega/vega-lite/issues/1386